src/main/java/org/primefaces/component/datagrid/DataGridRenderer.java

```diff
@@ -170,20 +171,24 @@ protected void encodeGrid(FacesContext context, DataGrid grid) throws IOExceptio
         int itemsToRender = rows != 0 ? rows : grid.getRowCount();
         int numberOfRowsToRender = (itemsToRender + columns - 1) / columns;
         int displayedItemsToRender = rowIndex + itemsToRender;
-        String columnClass = DataGrid.COLUMN_CLASS + " " + GridLayoutUtils.getColumnClass(columns);
+        String columnClass = DataGrid.COLUMN_CLASS + " " + GridLayoutUtils.getColumnClass(columns) + " " + grid.getRowStyleClass();
```

Comment:
sorry one more thing will `+ grid.getRowStyleClass` append the word "null" if its not set?  typically we check for `!LangUtils.isvalueblank` before appending an optional style class?

Reply:
Thanks for pointing me on this!